### PR TITLE
Please review code week0003.

### PR DESCRIPTION
**Pull request by hoangdangduy**
Hi,

I am newbie git and search on the internet to untracked anyfile when commit by git bash but I can't find solution. Can you guide for me. 
Thanks.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -7,4 +7,4 @@
 *.bat
 *.exe
 *.out
-*.class
+*.class
```
Comment (vietlq) [inline .gitignore]: Do not touch this file please! It meant for me to edit only.

File: week0003/hoangdangduy/java/Week0003.java [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+		String folderPath = null;
+		List<ModelCustomer> listAct = new ArrayList<>();
+
+		try {
+			folderPath = new java.io.File(".").getCanonicalPath();
+		} catch (IOException e1) {
+			e1.printStackTrace();
+		}
+
+		Path path = Paths.get(folderPath, "test_case_01.input");
```
Comment (vietlq) [inline week0003/hoangdangduy/java/Week0003.java]: The requirement was reading from STDIN, not from a file! There was a file redirection, not directly reading file in code. If you hard-code the file name, you won't be able to test with other test cases and this program will not work in production!

File: week0003/hoangdangduy/java/Week0003.java [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+	@Override
+	public String toString() {
+		return "ModelCustomer [action=" + action + ", name=" + name + ", money=" + money + "]";
+	}
+
+}
+
+public class Week0003 {
+
+	public List<ModelCustomer> readCustomerFromFile() {
```
Comment (vietlq) [inline week0003/hoangdangduy/java/Week0003.java]: Remove this function. Read from STDIN instead :)

File: week0003/hoangdangduy/java/Week0003.java [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+		default:
+			writeToFile("Invalid");
+			break;
+		}
+
+		return account;
+	}
+
+	public void writeToFile(String result) {
```
Comment (vietlq) [inline week0003/hoangdangduy/java/Week0003.java]: Remove this function. Write to STDOUT as required.

File: week0003/hoangdangduy/java/Week0003.java [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+
+		String folderPath = null;
+
+		try {
+			folderPath = new java.io.File(".").getCanonicalPath();
+			File file = new File(folderPath + "/test_case_01.output");
```
Comment (vietlq) [inline week0003/hoangdangduy/java/Week0003.java]: Do not write to a file!

File: week0003/hoangdangduy/java/Week0003.java [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+class ModelCustomer {
```
Comment (vietlq) [inline week0003/hoangdangduy/java/Week0003.java]: Why ModelCustomer contains an action? Please separate Client & Action into 2 separate classes.

File: week0003/hoangdangduy/java/test_case_01.output [hunk 7]
```diff
@@ -0,0 +1,12 @@
+True
+False
+True
+False
```
Comment (vietlq) [inline week0003/hoangdangduy/java/test_case_01.output]: Please remove this file.

**Comment by vietlq** [comment]
Hi @hoangdangduy, thanks for your feedback. I will have a video to show git flow ;)

File: .gitignore (hunk 1 above)
Comment (vietlq) [inline .gitignore]: Also revert this file!

File: week0003/hoangdangduy/java/Customer.java [hunk 8]
```diff
@@ -0,0 +1,35 @@
+public class Customer {
+	private String action;
```
Comment (vietlq) [inline week0003/hoangdangduy/java/Customer.java]: Customers must not know anything about actions! Actions must be put into another class!

File: week0003/hoangdangduy/java/Customer.java [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+		return money;
+	}
+
+	public void setMoney(double money) {
+		this.money = money;
+	}
+
+	@Override
+	public String toString() {
+		return "ModelCustomer [action=" + action + ", name=" + name + ", money=" + money + "]";
```
Comment (vietlq) [inline week0003/hoangdangduy/java/Customer.java]: We no longer have ModelCustomer

File: week0003/hoangdangduy/java/Week0003.java [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+
+		sc = new Scanner(System.in);
+
+		HashMap<String, Double> accounts = new HashMap<>();
+
+		while (true) {
+			String str = sc.nextLine();
+			String[] split = str.split(" ");
+			Customer customer = new Customer();
+			customer.setAction(split[0].toUpperCase());
```
Comment (vietlq) [inline week0003/hoangdangduy/java/Week0003.java]: There was no requirement to uppercase the action!

Comment (vietlq) [inline week0003/hoangdangduy/java/Week0003.java]: Also please give meaningful names instead of `split[*]`

**Comment by vietlq** [comment]
Yo! Are you still here? Please make changes so I can proceed.

**Comment by hoangdangduy** [comment]
I push new code from 2 days ago. Could you review and comment for me :D. When i push commit it have problem so I fix but it was removed two old commit :(

File: week0003/hoangdangduy/java/Process.java [hunk 11]
```diff
@@ -0,0 +1,84 @@
+import java.util.HashMap;
+
+public class Process {
+	private static final String add = "A";
```
Comment (vietlq) [inline week0003/hoangdangduy/java/Process.java]: The constants should be in uppercase so people can read easily:

`ADD/DEPOSIT/WITHDRAW`

File: week0003/hoangdangduy/java/Process.java [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+	
+	public HashMap<String, Double> process(String data, HashMap<String, Double> listAccount) {
+		Account account = new Account();
+		String[] inforTransaction = data.split(" ");
+		
+		switch (inforTransaction[0]) {
+		case add:
+			account = createAccount(inforTransaction[1], Double.parseDouble(inforTransaction[2]), listAccount);
+			return editListAccAfterAction(account, listAccount);
+		case deposite:
```
Comment (vietlq) [inline week0003/hoangdangduy/java/Process.java]: "Deposit", not "deposite"

File: week0003/hoangdangduy/java/Process.java [hunk 13]
```diff
@@ -0,0 +1,84 @@
...
+			return editListAccAfterAction(account, listAccount);
+		default:
+			Response response = new Response();
+			response.showMessage("Invalid");
+			break;
+		}
+		return listAccount;
+	}
+	
+	public Account createAccount(String name, Double money, HashMap<String, Double> listAccount){
```
Comment (vietlq) [inline week0003/hoangdangduy/java/Process.java]: Remember: The name is case-insensitive. So you can transform it to lowercase or uppercase before creating a new account.

Comment (vietlq) [inline week0003/hoangdangduy/java/Process.java]: Yep, it should affect all functions.

File: week0003/hoangdangduy/java/Process.java [hunk 14]
```diff
@@ -0,0 +1,85 @@
...
+
+public class Process {
+	private static final String ADD = "A";
+	private static final String DEPOSIT = "D";
+	private static final String WITHDRAW = "W";
+
+	public HashMap<String, Double> process(String data, HashMap<String, Double> listAccount) {
+		Account account = new Account();
+		String[] inforTransaction = data.split(" ");
+		inforTransaction[1] = inforTransaction[1].toUpperCase();
```
Comment (vietlq) [inline week0003/hoangdangduy/java/Process.java]: Parse double here. Give meaningful names: name, amount, action/command.

**Comment by vietlq** [comment]
Puuurfect!!! Finally you made it!

**Pull request merged**